src/products/utils/data.ts

```diff
@@ -276,3 +289,18 @@ export function mapFormsetStockToStockInput(
     warehouse: stock.id
   };
 }
+
+export const getEndPreorderDateInput = (
```

Comment:
Why not just use the whole formData as a prop and destruct inside of this function? We use it in couple of places and it'd save the refactor once the logic here grows or changes. 
```suggestion
export function getEndPreorderDateInput<T extends {
  hasPreorderEndDate: boolean,
  preorderEndDate?: string,
  preorderEndHour?: string
  }>({hasPreorderEndDate, preorderEndDate, preorderEndHour} : T)
```

And then 
```
getEndPreorderDateInput(formData)

Reply:
done